### PR TITLE
Update keyboard shortcuts menu button & dialog icon to a new 'keyboard' icon

### LINKED ISSUE #11767: Update keyboard shortcuts menu button & dialog icon to a new 'keyboard' icon
### Is your proposal related to a problem?

#11717 introduced a new keyboard shortcuts dialog as part of the work for https://github.com/wagtail/wagtail/issues/3949

For simplicity, we used an existing 'regex' icon for this, it would be nice to support a better icon.

Screenshot 2024-03-16 at 4 42 44 pm

### Describe the solution you'd like

1. Add a new `keyboard` icon to the core Wagtail admin icon set. (see instructions in the additional context).
2. Use this new icon in the sidebar menu `wagtail/admin/wagtail_hooks.py` see `register_keyboard_shortcuts_menu_item` for code.
3. Use this new icon in the dialog `wagtail/admin/templates/wagtailadmin/shared/keyboard_shortcuts_dialog.html`.

### Describe alternatives you've considered

* N/A - we should do this at some point, ideally before 6.1 goes out.

### Additional context

Here's the icon SVG (below), we must include the source reference, add this to the folder `wagtail/admin/templates/wagtailadmin/icons`.

Also ensure the icon is part of `wagtail/admin/wagtail_hooks.py` see `register_icons`, adding in alphabetical order.

For full instructions on adding icons see https://docs.wagtail.org/en/stable/contributing/ui_guidelines.html#adding-icons **Please follow these instructions exactly**.

There is no need for an RTL variation of this icon.

```svg
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--!Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2024 Fonticons, Inc.--><path d="M64 64C28.7 64 0 92.7 0 128V384c0 35.3 28.7 64 64 64H512c35.3 0 64-28.7 64-64V128c0-35.3-28.7-64-64-64H64zm16 64h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM64 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V240zm16 80h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V336c0-8.8 7.2-16 16-16zm80-176c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V144zm16 80h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V240c0-8.8 7.2-16 16-16zM160 336c0-8.8 7.2-16 16-16H400c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V336zM272 128h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H272c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM256 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H272c-8.8 0-16-7.2-16-16V240zM368 128h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H368c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM352 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H368c-8.8 0-16-7.2-16-16V240zM464 128h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM448 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V240zm16 80h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V336c0-8.8 7.2-16 16-16z"/></svg>
```

Screenshot 2024-03-16 at 4 54 16 pm
> From: https://fontawesome.com/icons/keyboard?f=classic&s=solid

### Working on this

* Anyone can contribute to this.
* View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.
* This aligns us closer to the goal designs in https://www.figma.com/proto/h67EsVXdWsfu38WGGxWfpi/Wagtail-CMS?type=design&node-id=14420-153315&viewport=6289%2C-12853%2C1&t=Sk9k7iALDZCA1bu7-0&scaling=scale-down-width&starting-point-node-id=14420%3A153315
* Here's a PR / commit that added some icons recently for reference 136a5486d8fab5f05c17f9ce86c9fc3308d5992d / #10320

### PR DESCRIPTION
**Pull request by rohitsrma**
Fixes #11767 

### After :

Screenshot (53)

Screenshot (54)

### CODE REVIEW

**Comment by lb-** [comment]
I think another step is still needed. 

> Go to the styleguide and copy the Wagtail icons table according to instructions in the template, pasting the result in wagtail_icons_table.txt.

https://docs.wagtail.org/en/stable/contributing/ui_guidelines.html#adding-icons


The rest of the code looks good, thank you. I'll do a proper review with the code checked out locally soon.

**Comment by rohitsrma** [comment]
Done!

screenshot-localhost_4000-2024 03 16-16_49_31

File: docs/_static/wagtail_icons_table.txt [hunk 1]
```diff
@@ -328,6 +328,11 @@
...
+<td><code>keyboard</code></td>
+<td> keyboard (solid): Font Awesome Pro 6.4.0 </td>
+<td><code>wagtailadmin/icons/keyboard.svg</code></td> </tr>
```
Comment (lb-) [inline docs/_static/wagtail_icons_table.txt:334]:
```suggestion
<td><code>wagtailadmin/icons/keyboard.svg</code></td>
</tr>
```

File: wagtail/admin/templates/wagtailadmin/icons/keyboard.svg [hunk 2]
```diff
@@ -0,0 +1 @@
...M64 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V240zm16 80h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V336c0-8.8 7.2-16 16-16zm80-176c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V144zm16 80h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V240c0-8.8 7.2-16 16-16zM160 336c0-8.8 7.2-16 16-16H400c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V336zM272 128h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H272c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM256 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H272c-8.8 0-16-7.2-16-16V240zM368 128h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H368c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM352 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H368c-8.8 0-16-7.2-16-16V240zM464 128h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM448 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V240zm16 80h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V336c0-8.8 7.2-16 16-16z"/></svg>
```
Comment (lb-) [inline wagtail/admin/templates/wagtailadmin/icons/keyboard.svg:1]:
```suggestion
<svg id="icon-keyboard" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><!--! keyboard (solid): Font Awesome Free 6.5.1--><path d="M64 64C28.7 64 0 92.7 0 128V384c0 35.3 28.7 64 64 64H512c35.3 0 64-28.7 64-64V128c0-35.3-28.7-64-64-64H64zm16 64h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM64 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V240zm16 80h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V336c0-8.8 7.2-16 16-16zm80-176c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V144zm16 80h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V240c0-8.8 7.2-16 16-16zM160 336c0-8.8 7.2-16 16-16H400c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H176c-8.8 0-16-7.2-16-16V336zM272 128h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H272c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM256 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H272c-8.8 0-16-7.2-16-16V240zM368 128h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H368c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM352 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H368c-8.8 0-16-7.2-16-16V240zM464 128h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16zM448 240c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V240zm16 80h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V336c0-8.8 7.2-16 16-16z"/></svg>
```

Just a note, we should clean up the viewbox as per the docs and this icon is not from the Pro font set but the free one.

**Review by lb- — APPROVED** [review]
Looking good thank you!

Thanks for the screenshots also, I have done a few small fixes (as per comments) & will push, check the docs and merge.

Really appreciate this.

File: wagtail/admin/templates/wagtailadmin/icons/keyboard.svg (hunk 2 above)
Comment (lb-) [reply wagtail/admin/templates/wagtailadmin/icons/keyboard.svg]: ignore this, I had it wrong. Looks like the viewbox was needed as is sorry.

**Comment by lb-** [comment]
@rohitsrma could you get the browser script working for you?

https://github.com/wagtail/wagtail/blob/24f5a4a14ce3df580ce5fc7f085f3e6a8357237e/wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html#L507

When I ran that I did not get the actual SVG content, only the svg use references.

I tried Firefox & Chrome, e.g.

```
<tr> 
<td><svg class="icon icon-keyboard w-w-8 w-h-8" aria-hidden="true"><use href="#icon-keyboard"></use></svg></td> 
<td><code>keyboard</code></td> 
<td> keyboard (solid): Font Awesome Free 6.5.1</td> 
<td><code>wagtailadmin/icons/keyboard.svg</code></td> </tr> 
```

Observe the use href= is there but not the actual SVG content.

**Comment by lb-** [comment]
Screenshot 2024-03-17 at 7 08 43 pm
Docs are working now (from when I broke it in the last push)

**Comment by lb-** [comment]
CI issues are not related this PR, merging as it all ran fine just before.

**Comment by rohitsrma** [comment]
> @rohitsrma could you get the browser script working for you?
> 
> https://github.com/wagtail/wagtail/blob/24f5a4a14ce3df580ce5fc7f085f3e6a8357237e/wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html#L507
> 
> When I ran that I did not get the actual SVG content, only the svg use references.
> 
> I tried Firefox & Chrome, e.g.
> 
> ```
> <tr> 
> <td><svg class="icon icon-keyboard w-w-8 w-h-8" aria-hidden="true"><use href="#icon-keyboard"></use></svg></td> 
> <td><code>keyboard</code></td> 
> <td> keyboard (solid): Font Awesome Free 6.5.1</td> 
> <td><code>wagtailadmin/icons/keyboard.svg</code></td> </tr> 
> ```
> 
> Observe the use href= is there but not the actual SVG content.

Thanks @lb- , Should I check this, or is it all sorted?.

**Comment by rohitsrma** [comment]
@lb- Yes i am also getting similar results for all icons.

```
<tr> 
<td><svg class="icon icon-folder-inverse w-w-8 w-h-8" aria-hidden="true"><use href="#icon-folder-inverse"></use></svg></td> 
<td><code>folder-inverse</code></td> 
<td> folder (solid): Font Awesome Pro 6.4.0 </td> 
<td><code>wagtailadmin/icons/folder-inverse.svg</code></td> </tr> 
<tr> 
```

**Comment by lb-** [comment]
Thanks @rohitsrma  - I have raised an issue for this https://github.com/wagtail/wagtail/issues/11771

No need to worry about it for now though, it's easy-ish enough to manually get the SVG content.

**Pull request merged**